From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: mm: make page_mapped_in_vma conditional on CONFIG_MEMORY_FAILURE
Date: Fri, 12 Apr 2024 20:35:01 +0100

This function is only currently used by the memory-failure code, so we can
omit it if we're not compiling in the memory-failure code.

Link: https://lkml.kernel.org/r/20240412193510.2356957-5-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Suggested-by: Miaohe Lin <linmiaohe@huawei.com>
Reviewed-by: Jane Chu <jane.chu@oracle.com>
Reviewed-by: Oscar Salvador <osalvador@suse.de>
Acked-by: Miaohe Lin <linmiaohe@huawei.com>
Cc: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_vma_mapped.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/page_vma_mapped.c~mm-make-page_mapped_in_vma-conditional-on-config_memory_failure
+++ a/mm/page_vma_mapped.c
@@ -314,6 +314,7 @@ next_pte:
 	return false;
 }
 
+#ifdef CONFIG_MEMORY_FAILURE
 /**
  * page_mapped_in_vma - check whether a page is really mapped in a VMA
  * @page: the page to test
@@ -344,3 +345,4 @@ unsigned long page_mapped_in_vma(struct
 out:
 	return pvmw.address;
 }
+#endif
_
